### ISSUE TITLE
'Inf' values in predictions, gives NaNs in probability maps

### ISSUE THREAD

**Issue by HolmlundWilliam** [MIC-DKFZ/nnUNet#1776]
Hi Fabian,

I've encountered an issue while using nnUNet (nnunetv2, version 2.2) for inference. When running predictions, I'm observing some 'NaN' voxel values in the output probability maps while the other voxel values behave as expected in the output probability masks.

I have been able to back-track the issue to the prediction, which contains 'Inf' values after this step:

_predict_from_raw_data.py_

`prediction = self.predict_logits_from_preprocessed_data(data).cpu()`

And then the NaNs appear after this step:

_export_prediction.py_

`predicted_probabilities = label_manager.apply_inference_nonlin(predicted_logits)`

Based on just looking at the outputs, it looks like the NaNs/Infs are where we would expect high probabilities, although not Inf! :)

Please let me know if you have any insights or suggestions for troubleshooting this issue. I would greatly appreciate your assistance in resolving this problem.

My setup for running inference:

```
`predictor = nnUNetPredictor(
    tile_step_size=0.5,
    use_gaussian=True,
    use_mirroring=True,
    perform_everything_on_gpu=True,
    device=torch.device('cuda', 0),
    verbose=False,
    verbose_preprocessing=False,
    allow_tqdm=True
)'

predictor.initialize_from_trained_model_folder("INPUT_FOLDER\nnUNetTrainer__nnUNetPlans__3d_fullres", use_folds=(0,), checkpoint_name='checkpoint_final.pth')

predictor.predict_from_files(INPUT',
                                            OUTPUT',
                                save_probabilities=True, overwrite=True,
                                num_processes_preprocessing=2, num_processes_segmentation_export=2,
                                folder_with_segs_from_prev_stage=None, num_parts=1, part_id=0)`
```

**Comment by FabianIsensee** [comment]
Hi @HolmlundWilliam , have you tried the most recent master? I have fixed this last year. Not sure if that fix made it into the latest release
Best,
Fabian

**Comment by HolmlundWilliam** [comment]
Hi @FabianIsensee , I tried with a newer version and it works!
Thank you!

**Comment by FabianIsensee** [comment]
Fantastic! We really should make this a new version soon
